python_modules/dagster/dagster/_config/pythonic_config/__init__.py

```diff
@@ -891,17 +891,46 @@ def _with_updated_values(
         )
         return out
 
-    def _resolve_and_update_env_vars(self) -> "ConfigurableResourceFactory[TResValue]":
-        """Processes the config dictionary to resolve any EnvVar values. This is called at runtime
-        when the resource is initialized, so the user is only shown the error if they attempt to
-        kick off a run relying on this resource.
+    def _resolve_and_update_configuration_values(
+        self, context: InitResourceContext
+    ) -> "ConfigurableResourceFactory[TResValue]":
+        """Processes the config dictionary attached to the resource to resolve any EnvVar values. As well
+        as overrides any values passed in via run_config (attached to the InitResourceContext).
+
+        This is called at runtime when the resource is initialized, so the user is only shown
+        the errors involving environment variable resolution if they attempt to kick off a
+        run relying on this resource.
 
         Returns a new instance of the resource.
         """
+        # This is the configuration passed via run_config (e.g. launchpad or Python API)
+
+        # Note right now we do not validate config, and that is a real problem. I added
+        # a test case in this PR to demonstrate in test_env_var_alongside_enum
+        #
+        # The problem is with enumerations. Process config converts dagster config enumerations
+        # to their underlying value. This in turn gets passed back to ConfigurableResourceFactory.__init__
+        # which in turn validates config again, and it fails because on the inbound Dagster config enums
+        # expect the NAME, not the VALUE.
+        #
+        # Effectively right now this codepath relies on the fact that you can passed validated
+        # and processed config back into ConfigurableResourceFactory.__init__ idempotently and
+        # enums break # that assumption. (Environment variable resolution for example works
+        # fine because the initial evaluation resolves the env var to its value, and subsequent
+        # evaluations are idempotent)
+
+        incoming_resource_config_evr = process_config(
+            self._schema.config_type, context.resource_config
+        )
+
+        # this is the data that was passed to the resource instance
         post_processed_data = _process_config_values(
```

Comment:
This call is a problem too because the "default" environment variables get evaluated. 

```
MyResource(foo=EnvVar("SHOULD_NOT_BE_ACCESS")
```